src/mlabns/mlabns.cpp

```diff
@@ -63,54 +51,7 @@ static ErrorOr<std::string> as_query(Settings &settings) {
 
 void query(std::string tool, Callback<Error, Reply> callback, Settings settings,
            Var<Reactor> reactor, Var<Logger> logger) {
-    ErrorOr<std::string> query = as_query(settings);
-    if (!query) {
-        callback(query.as_error(), Reply());
-        return;
-    }
-    std::string url = "https://mlab-ns.appspot.com/";
-    std::regex valid_tool("^[a-z]+$");
-    if (!std::regex_match(tool, valid_tool)) {
-        callback(InvalidToolNameError(), Reply());
-        return;
-    }
-    url += tool;
-    url += *query;
-    logger->info("query mlabns for tool %s", tool.c_str());
-    logger->debug("mlabns url: %s", url.c_str());
-    http::request("GET", url,
-                  [callback, logger](Error error, http::Response response) {
-                      if (error) {
-                          callback(error, Reply());
-                          return;
-                      }
-                      if (response.status_code != 200) {
-                          callback(UnexpectedHttpStatusCodeError(), Reply());
-                          return;
-                      }
-
-                      Reply reply;
-                      try {
-                          auto node = json::parse(response.body);
-                          reply.city = node["city"];
-                          reply.url = node["url"];
-                          for (auto ip2 : node["ip"]) {
-                              reply.ip.push_back(ip2);
-                          }
-                          reply.fqdn = node["fqdn"];
-                          reply.site = node["site"];
-                          reply.country = node["country"];
-                      } catch (std::invalid_argument &) {
-                          callback(JsonParseError(), Reply());
-                          return;
-                      } catch (std::out_of_range &) {
-                          callback(JsonKeyError(), Reply());
-                          return;
-                      }
-                      logger->info("mlabns says to use %s", reply.fqdn.c_str());
-                      callback(NoError(), reply);
-                  },
-                  {}, "", settings, logger, reactor);
+    query_debug(tool, callback, settings, reactor, logger);
```

Comment:
I'd rename `query_debug` as `query_impl`, because that template is the *actual* implementation of the function, and instead I think that the name `query_debug` is a bit misleading. (This is how things are done in the NDT branch.)